src/Service/Helpers/UnknownCredentialException.cs

```diff
@@ -0,0 +1,16 @@
+namespace Passwordless.Service.Helpers;
+
+public class UnknownCredentialException : ApiException
+{
+    public UnknownCredentialException(string message) : base(message)
+    {
+    }
+
+    public UnknownCredentialException(string message, int statusCode, Dictionary<string, object> extras = null) : base(message, statusCode, extras)

```

Comment:
Personally for each exception that wraps `ApiException` I would have it not simply expose the base class constructors and instead be like:
```C#
public UnknownCredentialException(string credentialId)
: base("unknown_credential", "We don't recognize the passkey you sent us", 400, new () { { "credentialId", credentialId }})
{
    CredentialId = credentialId;
}
```
Maybe it could have another constructor that allows customizing the message but for the most part it is a helper around `ApiException`. Let me know if you disagree though.